docs/good-practice/update/update.md

```diff
@@ -0,0 +1,92 @@
+# How to ensure safe update ?
+
+To manage update you need to split this operation in two. An update is a composition of an adding and a deletion.
+
+## Rename property/column
+
+Here the scenario if you want to rename property/column 'name' into 'lastname' :
+- In V+1 :
+  - create a BEFORE script that add a property/column 'lastname' with null value/constraint.
+  - update the code to write both on 'name' and 'lastname'.
+    - do not read the new property/column 'lastname'. Keep reading for 'name'.
+- In V+2 :
+  - create a BEFORE script that copy 'name' value on 'lastname' for entries having 'lastName' null.
+  - update the code to read and write only 'lastname'.
+    - so 'name' becomes unnecessary.
+  - create an AFTER script that remove the property/column 'name'.
+  - if your db has schema and 'lastname' is mandatory, create an AFTER script to add non-null constraint.
+  
+**Note :** if your db manages triggers, you can use them to ensure both column are synchronized.
+
+## Rename value
+
+Here the scenario if you want to rename possible value from 'ENDED' into 'DONE' :
+- In V+1 :
+  - create a BEFORE script that add 'DONE' as value in your schema.
+  - add the value 'DONE' as a possible value in your code (you maybe have an enum to manage these possible values).
+    - do not use the 'DONE' value for now.
+- In V+2 :
+  - Update your code to write only 'DONE' in db
+    - you need to continue to manage 'ENDED' as possible value on read
+- In V+3 :
+  - create a BEFORE script to update all entries using 'ENDED' to replace it with 'DONE'.
+  - update the code to remove 'ENDED' value.
+  - if your db has schema, create an AFTER script to remove 'ENDED' as possible value.
+  
+
+## Update type
+
+To proceed to an update type we can perform :
+- add a new temporary property from the one we want to rename
+- remove the original property
```

Comment:
Is there a need to remove the property? I did such an operation on a mongo database and renaming my temporary property was enough to replace the old value with my new one.

Reply:
Actually I read it all and you did take explain everything so I'm resolving this conversation.